lib/core/versions/latest/Operation.ts

```diff
@@ -280,22 +243,49 @@ export default class Operation {
       throw new SidetreeError(ErrorCode.OperationPayloadMissingOrIncorrectType);
     }
 
-    // Verify operation specific payload schema.
-    switch (operationType) {
+    // Initialize common operation properties.
+    this.type = decodedPayload.type;
+    this.signingKeyId = decodedProtectedHeader.kid;
+    this.encodedProtectedHeader = operation.protected;
+    this.encodedPayload = operation.payload;
+    this.signature = operation.signature;
+    this.operationHash = Operation.computeHash(this.encodedPayload);
+
+    // Verify operation specific payload schema and further decode if needed, then initialize.
+    switch (this.type) {
       case OperationType.Create:
+        // additional parsing required because did doc is nest base64url encoded
+        decodedPayload.didDocument = JSON.parse(Encoder.decodeAsString(decodedPayload.didDocument));
```

Comment:
This is a fix for did doc parsing as we added another layer of encoding and wasn't handling it.